XenAdmin/Alerts/Types/CssExpiryAlert.cs

```diff
@@ -0,0 +1,108 @@
+﻿/* Copyright (c) Cloud Software Group, Inc. 
+ * 
+ * Redistribution and use in source and binary forms, 
+ * with or without modification, are permitted provided 
+ * that the following conditions are met: 
+ * 
+ * *   Redistributions of source code must retain the above 
+ *     copyright notice, this list of conditions and the 
+ *     following disclaimer. 
+ * *   Redistributions in binary form must reproduce the above 
+ *     copyright notice, this list of conditions and the 
+ *     following disclaimer in the documentation and/or other 
+ *     materials provided with the distribution. 
+ * 
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND 
+ * CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, 
+ * INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF 
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE 
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR 
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, 
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, 
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR 
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS 
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, 
+ * WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING 
+ * NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE 
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF 
+ * SUCH DAMAGE.
+ */
+
+using System;
+using System.Diagnostics;
+using System.Xml;
+using XenAPI;
+
+
+namespace XenAdmin.Alerts
+{
+    public class CssExpiryAlert : MessageAlert
+    {
+        private readonly DateTime? _cssExpiryDate;
+        private readonly string _title;
+        private readonly string _description;
+
+        public CssExpiryAlert(Message m)
+            : base(m)
+        {
+            try
+            {
+                var doc = new XmlDocument();
+                doc.LoadXml(m.body);
+                var nodes = doc.GetElementsByTagName("date");
+
+                if (nodes.Count > 0 && Util.TryParseIso8601DateTime(nodes[0].InnerText, out var result))
+                    _cssExpiryDate = result;
+            }
+            catch
+            {
+                //ignore
+            }
+
+            switch (Message.Type)

```

Comment:
nit do you think we should add a `default`? there's a few `Message.Type` we're not supporting here

Reply:
Static analysers will flag this, in reality though it is not needed because the constructor of the class is only called for the listed message types, hence  we don't expect any surprises (compare with the ParseMessage method calling the constructor, where indeed we need the default clause for unknown messages).